site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Copy-on-Write and Merge-on-Read
+
+This page explains the concept of copy-on-write and merge-on-read in the context of Iceberg to provide readers more clarity around Iceberg's table spec design.
+
+## Introduction
+
+In Iceberg, copy-on-write and merge-on-read are different ways to handle row-level update and delete operations. Here are their definitions:
+
+- **copy-on-write (CoW)**: an update/delete directly rewrites the entire affected data files.
+- **merge-on-read (MoR)**: update/delete information is encoded in the form of delete files. The table reader can apply all delete information at read time. A compaction process takes care of merging delete files into data files asynchronously. 
+
+Clearly, CoW is more efficient in reading data, but MoR is more efficient in writing data.
```

Comment:
Would just skip the word Clearly here, computers are hard so it may not be clear :)

Perhaps something like

Copy on write provides better read performance:

Copy on write provides better performance on reading because no additional files need to be read and merged to get the current state of the table but this comes at the cost of worse performance at write time. At write time Copy on Write must rewrite an entire file even if only a single row is changed within that file. Data was previously written and unmodified still must be rewritten if any adjacent row was modified.

Merge on Read provides better write performance:

Merge on Read only needs to write new files with the data that has changed in the table which means writing significantly less information than Copy on Write. Write performance is increased, but every read now requires reading not just the requested data files but also all delete files which apply to those data files.